### PR TITLE
Add Dictation and WindowHasher extensions

### PR DESCRIPTION
**Pull request by pooiod**
An extension to allow you to use your speech as text
It mostly only works with Chrome though, and also Safari somehow

it comes with 4 blocks:
image
Dictation

Also Window Hasher

### CODE REVIEW

**Comment by RedMan13** [comment]
mmmmaby, @JeremyGamer13 whab tdo yo think

**Comment by JeremyGamer13** [comment]
- speech is spelled as "speach" in some parts
- this api is [not widely supported,](https://caniuse.com/speech-recognition) you might want to add a message somewhere about that

**Comment by pooiod** [comment]
Does the "can use" block count?

File: static/extensions/pooiod/Dictation.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  'use strict';
+
+  if (!Scratch.extensions.unsandboxed) {
+    throw new Error('Your microphone can\'t be accessed in the sandbox');
+  }
+
+  class dictation {
+    constructor() {
+      this.recognizedSpeech;
+      this.cando = Scratch.vm.runtime.isPackaged;
```
Comment (SharkPool-SP) [inline static/extensions/pooiod/Dictation.js:13]: Dont use `runtime.isPackaged` in this use.

Use `typeof scaffolding === "undefined"`. It returns true if its not packaged

**Comment by JeremyGamer13** [comment]
> Does the "can use" block count? Also I fixed the spelling

some browsers have the class but dont fire the events apparently, but i think that'll work fine enough if you dont want to figure out which browsers support it completely

File: static/extensions/pooiod/Dictation.js (hunk 1 above)
Comment (RedMan13) [reply static/extensions/pooiod/Dictation.js]: why do that? isPackaged should just be yes if packaged runtime no if unpackaged?

Comment (JeremyGamer13) [reply static/extensions/pooiod/Dictation.js]: packaged runtime can be disabled in the packager (its very confusing)

Comment (SharkPool-SP) [reply static/extensions/pooiod/Dictation.js]: > why do that? isPackaged should just be yes if packaged runtime no if unpackaged?

Long story short, isPackaged is kind of misinformation. It doesn't report true if it's in a packaged environment. It returns true depending on certain packager information. 

For example, if you disable the "Remove Assets for RAM" option, isPackaged will return false

**Comment by pooiod** [comment]
I accidentally added WindowHasher to this pull request, but I'm committing to it
WindowHasher
image

**Comment by RedMan13** [comment]
2harts window hasher

**Comment by SharkPool-SP** [comment]
what is window hashing supposed to do?

**Comment by dumorando** [comment]
> 2harts window hasher

the thing after # in a url i thinnk

**Comment by SharkPool-SP** [comment]
fix the conflicts and ill fix up the rest and merge it

**Comment by CodeFoxy-Github** [comment]
i tryed the window hasher it might can't be exploit so lgtm

File: static/extensions/pooiod/Dictation.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+          },
+        ]
+      };
+    }
+
+    WaitAndrecognizeSpeech() {
+      return this.recognizeSpeech(false);
+    }
+
+    canuse() {
```
Comment (CodeFoxy-Github) [inline static/extensions/pooiod/Dictation.js:52]: **How about can use are checked in imported**Edit: bruh no

**Pull request merged**